resources/web/map/js/trajectory.js

```diff
@@ -0,0 +1,76 @@
+var data = {
+    type: 'Feature',
+    properties: {},
+    geometry: {
+        type: 'LineString',
+        coordinates: []
+    }
+};
+
+const style = {
+    "version": 8,
+    "sources": {
+        "osm": {
+            "type": "raster",
+            "tiles": ["https://a.tile.openstreetmap.org/{z}/{x}/{y}.png"],
+            "tileSize": 256,
+            "attribution": "&copy; OpenStreetMap Contributors",
+            "maxzoom": 22
+        },
+        "route": {
+            "type": "geojson",
+            data
+        }
+    },
+    "layers": [
+        {
+            "id": "osm",
+            "type": "raster",
+            "source": "osm"
+        },
+        {
+            "id": "route",
+            "type": "line",
+            "source": "route",
+            "layout": {
+                "line-join": "round",
+                "line-cap": "round"
+            },
+            "paint": {
+                "line-color": "#ff8600",
+                "line-width": 10
+            }
+        }
+    ]
+};
+
+var map = new maplibregl.Map({
+    container: 'map',
+    style,
+    center: [-122.486052, 37.830348],  // Initial focus coordinate
+    zoom: 16
+});
+
+// MapLibre GL JS does not handle RTL text by default, so we recommend adding this dependency to fully support RTL rendering.
+maplibregl.setRTLTextPlugin('https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-rtl-text/v0.2.1/mapbox-gl-rtl-text.js');
+
+// Add zoom and rotation controls to the map.
+map.addControl(new maplibregl.NavigationControl());
+
+map.on('load', function () {
+    console.log("loaded");
+    var start;
+    new QWebChannel(qt.webChannelTransport, (channel) => {
+        channel.objects.currPos.recvPos.connect((id, lat, lng) => {
```

Comment:
should be a bug here, we only send diffs but we also pause the diffs hence if we are not on solution tab, this would not track that location.
*hypothetically*

should adjust to send state diffs as per https://swift-nav.atlassian.net/browse/CPP-826

Reply:
also todo: extend id on different layers for colour coding.